### PR TITLE
Rename Rect3 to AABB.

### LINKED ISSUE #12973: Rename Rect3 to AABB
This has been on my mind since @reduz renamed AABB to Rect3, and I'd like to discuss this before beta, while we can still make such a change.

This change was made along Matrix3 -> Basis change (which made sense because Matrix3 was not a general matrix but a rotation-scale matrix). but AABB -> Rect3 change did the opposite thing: AABB is a special kind of Rect3, the name should have been kept (since axis-aligned bounding box is a very specific case of a 3D generalization of rectangle.

So I'd like to rename Rect3 back to AABB.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #12973.

### CODE REVIEW

**Comment by HummusSamurai** [comment]
https://github.com/godotengine/godot/issues/12973

File: core/math/aabb.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*************************************************************************/
-/*  rect3.cpp                                                            */
+/*  aabb.cpp                                                            */
```
Comment (akien-mga) [inline core/math/aabb.cpp]: There's one additional space needed to align the `*/` at the end of the line :P

File: core/math/aabb.h [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*************************************************************************/
-/*  rect3.h                                                              */
+/*  aabb.h                                                              */
```
Comment (akien-mga) [inline core/math/aabb.h]: Same here.

File: core/variant.h [hunk 3]
```diff
@@ -136,7 +136,7 @@ class Variant {
 		int64_t _int;
 		double _real;
 		Transform2D *_transform2d;
-		Rect3 *_rect3;
+		::AABB *_aabb;
```
Comment (akien-mga) [inline core/variant.h]: I wonder if the Variant enum was not named `_AABB` in the past to avoid this naming collision. But maybe this solution is better.

Comment (reduz) [reply core/variant.h]: I agree the solution is better

Comment (tagcup) [reply core/variant.h]: Wow it's good as-is then?

Comment (tagcup) [reply core/variant.h]: Couldn't be sure myself

File: core/math/aabb.cpp (hunk 1 above)
Comment (tagcup) [reply core/math/aabb.cpp]: Ah so clang-format can't detect those

**Comment by tagcup** [comment]
OK, done

**Comment by reduz** [comment]
PR seems good, but something is wrong with Travis

**Comment by tagcup** [comment]
Ah, yeah, that'd be my clang-format... Arch Linux has version 5.0.0 which seems to do weird things, just left them alone to make it happy, but the version on the travis is apparently different and likes to do things differently

**Comment by akien-mga** [comment]
Yeah for now I'm ignoring those clang-format issues, version 3.9 and 5.0.0 behave differently regarding comments around defines/macros... I'll soon up our requirements to 5.0.0 so that we all have the same version.

**Comment by gdeer81** [comment]
I'm following the "build Godot with mono" instructions and I get this error:
```
modules/mono/mono_gd/gd_mono_field.cpp: In member function 'void GDMonoField::set_value(MonoObject*, const Variant&)':
modules/mono/mono_gd/gd_mono_field.cpp:133:31: error: expected type-specifier before 'AABB'
     SET_FROM_STRUCT_AND_BREAK(AABB);
                               ^
... stacktrace truncated...
scons: *** [modules/mono/mono_gd/gd_mono_marshal.x11.tools.64.mono.o] Error 1
scons: building terminated because of errors.
```
When I reverted this commit the build worked fine.

**Comment by Valentactive** [comment]
I can confirm the same problem as @gdeer81 with in the first step of building with mono module enabled.

**Comment by tagcup** [comment]
Haven't see this problem with latest clang++. Which compiler are you using?

**Comment by Valentactive** [comment]
I made that build with clang 5.0.1 (since 5.0 is actually considered the stable one). I was on 6.0 until two weeks ago but I removed it since their repo for ubuntu (16.04) didn't work for me a few days and I was annoyed by the errors when updating my machine. I can give later or tomorrow a try with 6.0 again.

**Comment by neikeq** [comment]
@tagcup Although #13016 fixes these compiler errors, `Variant::Type::AABB` should be renamed to `Variant::Type::_AABB`.

**Comment by tagcup** [comment]
I have no opinion either way, maybe it's better to do that change as a part of #13016

**Comment by Valentactive** [comment]
This patch solves the problem (if any confirmation is needed )

File: core/variant_parser.cpp [hunk 4]
```diff
@@ -595,7 +595,7 @@ Error VariantParser::parse_value(Token &token, Variant &value, Stream *p_stream,
 			value = Quat(args[0], args[1], args[2], args[3]);
 			return OK;
 
-		} else if (id == "Rect3" || id == "AABB") {
```
Comment (reduz) [inline core/variant_parser.cpp]: This was a pretty bad idea, renders existing scenes incompatible. Will add a fix asap.

Comment (tagcup) [reply core/variant_parser.cpp]: Ah, that's what is was for, sorry

**Pull request merged**